src/main/groovy/lang/ObjectRange.java

```diff
@@ -261,16 +260,12 @@ public T get(int index) {
     @Override
     public boolean containsWithinBounds(Object value) {
         if (value instanceof Comparable) {
-            final int result = compareTo(from, (Comparable) value);
-            return result == 0 || result < 0 && compareTo(to, (Comparable) value) >= 0;
+            final int result = compareTo(from, value);
+            return result == 0 || result < 0 && compareGreaterThanEqual(to, value);
         }
         return contains(value);
     }
 
-    protected int compareTo(Comparable first, Comparable second) {
```

Comment:
An earlier request was to make compareTo protected to allow subclasses of ObjectRange to change the behavior. By optimizing this we have taken away that capability. But the question is, was the real motivation behind the previous request so that they could optimize the method themselves? If that is the case, we are in a better position.